### PR TITLE
[#1347] Adapt the device registry guide to the new HTTP API

### PR DESCRIPTION
**Pull request by dejanb**

### CODE REVIEW

File: site/documentation/content/user-guide/device-registry.md [hunk 1]
```diff
@@ -8,84 +8,83 @@ The Device Registry component provides exemplary implementations of Hono's [Tena
...
 ## Managing Tenants
 
 The following sections describe the resources representing the operations of the Tenant API and how they can be used to manage tenants.
-Please refer to the [Tenant API]({{< relref "/api/Tenant-API.md" >}}) for the specific elements that are explained in detail there.
+Please refer to the [Device Registry HTTP API]({{< relref "/api/device-registry-v1.yaml" >}}) for the specific elements that are explained in detail there.
 
 ### Add a Tenant
 
-* URI: `/tenant`
+* URI: `/tenants/${tenantId}`
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: `/v1/tenants/...`?

File: site/documentation/content/user-guide/device-registry.md [hunk 2]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
-  * (required) A JSON object as specified by [Credentials Format]({{< relref "/api/Credentials-API.md#credentials-format" >}}) of the Credentials API.
+  * (required) A JSON object as specified by [Credentials schema] ({{< relref "/api/device-registry-v1.yaml" >}}) of the HTTP API specification.
 * Status Codes:
-  * 201 (Created): Credentials have been added successfully under the resource indicated by `Location` header.
+  * 204 (No Content): Credentials have been updated successfully under the resource indicated by `Location` header.
   * 400 (Bad Request): The credentials have not been added because the request was malformed, e .g. because the payload did not contain required values. The response body may contain hints regarding the cause of the problem.
-  * 409 (Conflict): Credentials of the given type for the given *auth-id* already exist for the tenant. The request has not been processed.
 * Response Headers:
   * `Location`: The URI under which the newly created resource can be accessed.
+  * `ETag`: Version of the resource
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: there is no *location* and no *etag* header in response to a `PUT` request

File: site/documentation/content/user-guide/device-registry.md [hunk 3]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
+* URI: `/v1/credentials/${tenantId}/${deviceId}`
+* Method: `PUT`
 * Request Headers:
   * (required) `Content-Type`: `application/json` (no other type supported)
 * Request Body:
-  * (required) A JSON object as specified by [Credentials Format]({{< relref "/api/Credentials-API.md#credentials-format" >}}) of the Credentials API.
+  * (required) A JSON object as specified by [Credentials schema] ({{< relref "/api/device-registry-v1.yaml" >}}) of the HTTP API specification.
 * Status Codes:
-  * 201 (Created): Credentials have been added successfully under the resource indicated by `Location` header.
+  * 204 (No Content): Credentials have been updated successfully under the resource indicated by `Location` header.
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *location* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 4]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
       }]
-    }' http://localhost:28080/credentials/DEFAULT_TENANT
+    }]' http://localhost:28080/v1/credentials/DEFAULT_TENANT/4710
       
-    HTTP/1.1 201 Created
-    Location: /credentials/DEFAULT_TENANT/sensor10/hashed-password
-    Content-Length: 0
+    HTTP/1.1 204 No Content
+    Location: /v1/credentials/DEFAULT_TENANT/4710
+    ETag: becc93d7-ab0f-48ec-ad26-debdf339cbf4x
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *location* nor *etag* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 5]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
             "pwd-hash": "'$PWD_HASH'"
         }]
-      }' http://localhost:28080/credentials/DEFAULT_TENANT
+      }]' http://localhost:28080/v1/credentials/DEFAULT_TENANT/4720
     
-    HTTP/1.1 201 Created
-    Location: /credentials/DEFAULT_TENANT/sensor20/hashed-password
-    Content-Length: 0
+    HTTP/1.1 204 No Content
+    Location: /v1/credentials/DEFAULT_TENANT/4720
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *location* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 6]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
+    ETag: 02c99fb5-af8c-409f-8520-b405e224b27f
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *etag* in response

File: site/documentation/content/user-guide/device-registry.md [hunk 7]
```diff
@@ -160,52 +161,53 @@ The following sections describe the resources representing the operations of the
...
+    }' http://localhost:28080/v1/devices/DEFAULT_TENANT/4711
 
 The response will contain a `Location` header containing the resource path created for the device. In this example it will look
 like this:
 
     HTTP/1.1 201 Created
-    Location: /registration/DEFAULT_TENANT/4711
-    Content-Length: 0
+    Location: /v1/devices/DEFAULT_TENANT/4711
+    ETag: becc93d7-ab0f-48ec-ad26-debdf339cbf4
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *etag* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 8]
```diff
@@ -117,39 +116,41 @@ The following command disables the MQTT adapter for devices that belong to the t
               "type" : "hono-mqtt",
               "enabled" : true
               } ]
-      }' http://localhost:28080/tenant/tenantMqttAdapter
+      }' http://localhost:28080/v1/tenants/tenantMqttAdapter
     
     HTTP/1.1 204 No Content
+    ETag: 8919c736-30aa-40ce-a45a-830b90c4cd42
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *etag* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 9]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
+        "secrets": [{
+            "hash-function" : "sha-512",
+            "pwd-hash": "'$PWD_HASH'",
+            "not-after": "2018-01-01T00:00:00+01:00"
+        }]
+    }' http://localhost:28080/v1/credentials/DEFAULT_TENANT/4720
+
+    HTTP/1.1 204 No Content
+    Location: /v1/credentials/DEFAULT_TENANT/4720
+    ETag: becc93d7-ab0f-48ec-ad26-debdf339cbf4x
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *location* no *etag* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 10]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
-### Get Credentials by Authentication Identifier and Type
-
-* URI: `/credentials/${tenantId}/${authId}/${type}`
-* Method: `GET`
-* Status Codes:
-  * 200 (OK): Credentials for the given parameters have been found. The response body will contain credentials as specified by [Credentials Format]({{< relref "/api/Credentials-API.md#credentials-format" >}}) of the Credentials API.
-  * 404 (Not Found): No credentials for the given parameters are registered for the given tenant.
+    HTTP/1.1 204 No Content
+    Location: /v1/credentials/DEFAULT_TENANT/4720
+    ETag: 122c971a-505a-4336-8f7d-640360e909bc
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *location* nor *etag* header in response

File: site/documentation/content/user-guide/device-registry.md [hunk 11]
```diff
@@ -214,278 +216,193 @@ The following command retrieves registration data for device `4711`:
...
-                "pwd-hash": "tnxz0zDFs+pJGdCVSuoPE4TnamXsfIjBEOb0rg3e9WFD9KfbCkoRuwVZKgRWInfqp87kCLsoV/HEwdJwgw793Q=="
-            }
-        ],
-        "type": "hashed-password"
-    }
+    curl -i -X PUT -H 'Content-Type: application/json' --data-binary '[]' http://localhost:28080/v1/credentials/DEFAULT_TENANT/4710
 
+    HTTP/1.1 204 No Content
+    Location: /v1/credentials/DEFAULT_TENANT/4710
+    ETag: d383ba4d-1853-4d03-b322-b7ff5af05ae2
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: no *location* nor *etag* header in response

File: site/documentation/content/user-guide/device-registry.md (hunk 2 above)
Comment (sophokles73) [reply site/documentation/content/user-guide/device-registry.md]: there is definitely no *location* header in a response to `PUT` ...

File: site/documentation/content/user-guide/device-registry.md [hunk 12]
```diff
@@ -214,278 +216,187 @@ The following command retrieves registration data for device `4711`:
...
+    curl -i http://localhost:28080/v1/credentials/DEFAULT_TENANT/4710
+
     HTTP/1.1 200 OK
     Content-Type: application/json; charset=utf-8
-    Content-Length: 222
 
-    {
-      "total": 1,
-      "credentials": [
+    [{
```
Comment (sophokles73) [inline site/documentation/content/user-guide/device-registry.md]: IMHO  the response also contains an *etag* header here

Comment (dejanb) [reply site/documentation/content/user-guide/device-registry.md]: yup

**Pull request merged**